src/number/format-decimal.js

```diff
@@ -19,6 +19,15 @@ module.exports = function(value, localeData, options) {
 	var ret = '';
 	var processedDecimal = (valueStr.indexOf('.') < 0);
 	var digitNum = 0;
+	var currentGroupSize ;
+	var groupSizes;
+	if ( Array.isArray( localeData.groupSize) ) {
+		groupSizes = localeData.groupSize.slice(); 	// make a copy of the groupsizes
+		currentGroupSize = groupSizes.shift();		// pop the first groupSize from the front
+	} else {
+		groupSizes = [];
+		currentGroupSize = localeData.groupSize;
+	}
```

Comment:
Might be cleaner to just default to `[localeData.groupSize]`? e.g. these 8 lines could become:

    var groupSizes = Array.isArray(localeData.groupSizes) ?
      localeData.groupSizes.slice() : [localeData.groupSizes];
    var currentGroupSize = groupSizes.shift();